test/mode-test.c

```diff
@@ -135,7 +135,7 @@ END_TEST
 
 START_TEST(test_mode_result)
 {
-    char *res;
+    const char *res = "";
```

Comment:
Not sure if const is correct here, being that the 3rd argument of mode_result is not const.

Reply:
Yes, that is true, but `""` is a `const char *`, modifying it results in undefined behavior. I think the third argument of mode_result should be const, because it's an input string. (That's what I meant by I've started adding const qualifiers).

Reply:
I initialized it to be modifiable. I still this is wrong, but it's consistent with the signature of `mode_result`.